### ISSUE TITLE
蒸馏后学生模型乱码

### ISSUE THREAD

**Issue by peixin-lin** [airaria/TextBrewer#28]
你好，我参考cmrc2018_example做了一个QA的bert三层蒸馏，但是蒸馏后学生模型的输出却乱码了，请问这种情况的原因一般是什么呢，是代码的问题还是调参的问题呢，我的超参数选择了推荐的lr: 1e-4和epoch: 50，macthes选择了'L3_hidden_smmd',  'L3_hidden_mse', 'L3_attention_mse', 'L3_attention_ce', 'L3_attention_mse_sum', 'L3_attention_ce_mean'。

**Comment by airaria** [comment]
输出乱码是指？是解码结果完全错误，但还是passage中的一个span吗？

**Comment by peixin-lin** [comment]
> 输出乱码是指？是解码结果完全错误，但还是passage中的一个span吗？还是解码结果完全不是一个原文中的span?

一开始是输出了一些无意义的字符，后来调大学习率之后编程重复输出同一个汉字，并且不同的输入会是同样的输出（跟老师模型表现差异很大），请问你们的学习率和epoch是用的什么数值，或者是有其他哪个重要的超参数影响比较大吗

**Comment by airaria** [comment]
learning rate = 1e-4
epochs = 30
temperature = 8
不使用matches，用BasicDistiller就可以得到正常的结果。

不过不像是超参设置原因，更像是数据前处理或解码不当，你可以检查下数据预处理、模型输出（logits和loss）和adaptor定义

**Comment by peixin-lin** [comment]
> learning rate = 1e-4
> epochs = 30
> temperature = 8
> 不使用matches，用BasicDistiller就可以得到正常的结果。
> 
> 不过不像是超参设置原因，更像是数据前处理或解码不当，你可以检查下数据预处理、模型输出（logits和loss）和adaptor定义=

数据处理和模型解码方式老师模型和学生模型是一致的，老师模型的表现是正常的，是蒸馏后的学生模型才表现异常，adaptor的定义是没问题的。
不知道是不是任务不同的原因，我的任务是一个闲聊对话的生成任务，即输入只有问题然后输出一个回答的形式，跟squad的数据集还是差别比较大，请问你们有做过相关的工作吗？

**Comment by airaria** [comment]
> > learning rate = 1e-4
> > epochs = 30
> > temperature = 8
> > 不使用matches，用BasicDistiller就可以得到正常的结果。
> > 不过不像是超参设置原因，更像是数据前处理或解码不当，你可以检查下数据预处理、模型输出（logits和loss）和adaptor定义=
> 
> 数据处理和模型解码方式老师模型和学生模型是一致的，老师模型的表现是正常的，是蒸馏后的学生模型才表现异常，adaptor的定义是没问题的。
> 不知道是不是任务不同的原因，我的任务是一个闲聊对话的生成任务，即输入只有问题然后输出一个回答的形式，跟squad的数据集还是差别比较大，请问你们有做过相关的工作吗？

抱歉，没有做过相关工作。
会不会和kd_loss的计算方式不对有关系？比如，最近发现TextBrewer无法处理多标签分类任务。
所以我的意思是，有可能原先代码中的kd_loss的计算方式并没有考虑到你们的任务输出格式。
当然你们具体的任务设置的我也不清楚，我也只能做些猜测。
